homeassistant/components/tellduslive/__init__.py

```diff
@@ -72,15 +71,11 @@
     hass.data[DATA_CONFIG_ENTRY_LOCK] = asyncio.Lock()
     hass.data[CONFIG_ENTRY_IS_SETUP] = set()
 
-    client = TelldusLiveClient(hass, entry, session)
+    interval = entry.data[KEY_SCAN_INTERVAL]
+    _LOGGER.debug('Update interval %s seconds.', interval)
+    client = TelldusLiveClient(hass, entry, session, interval)
     hass.data[DOMAIN] = client
-    await async_add_hubs(hass, client, entry.entry_id)
-    hass.async_create_task(client.update())
-
-    interval = timedelta(seconds=entry.data[KEY_SCAN_INTERVAL])
-    _LOGGER.debug('Update interval %s', interval)
-    hass.data[INTERVAL_TRACKER] = async_track_time_interval(
-        hass, client.update, interval)
+    hass.loop.create_task(async_add_hubs(hass, client, entry.entry_id))
```

Comment:
Maybe we should rename `async_add_hubs` now that it also schedules a client update?

Reply:
We should save the returned task, and cancel it if not done, when home assistant shuts down or the entry is unloaded.

Reply:
@MartinHjelmare, should I make this part of `TelldusLiveClient.__init__`? (can an `__init__` be async)?